src/core/lib/security/security_connector/security_connector.cc

```diff
@@ -1157,3 +1101,77 @@ grpc_security_status grpc_ssl_server_security_connector_create(
   }
   return retval;
 }
+
+namespace grpc_core {
+
+tsi_ssl_root_certs_store* DefaultSslRootStore::default_root_store_ = nullptr;
+grpc_slice DefaultSslRootStore::default_pem_root_certs_;
```

Comment:
Why do we need this again?

Reply:
This is because OpenSSL 1.0x  or 0.9 does not support sharing of X509_STORE.  
https://www.openssl.org/docs/man1.1.0/crypto/X509_STORE_up_ref.html

When we pass X509_STORE to SSL_CTX, it also gives ownership. In OpenSSL 1.1, we can increase the reference counter to make sure when SSL_CTX destroys, shared X509_STORE does not destroy.

So I decide to not support caching default SSL root store X509_STORE, if users use an older version of OpenSSL. Thus, we also need to pass the default_pem_root_certs char* as well, in case someone uses an older version of OpenSSL.

Reply:
If we want to avoid this default_pem_root_certs grpc_slice, we can save this in tsi_ssl_root_certs_store.